Frontend/android-native/app/src/main/java/in/juspay/mobility/MainActivity.java

```diff
@@ -263,6 +265,108 @@ public void onBackPressed() {
         }
     }
 
+    public String getDeviceRAM()
+    {
+        String deviceRAM = sharedPref.getString("DEVICE_RAM", "__failed");
+
+        if(deviceRAM != "__failed")
+            return deviceRAM;
+
+        long memory=0;
+
+        try {
+            ActivityManager activityManager = (ActivityManager) getApplicationContext().getSystemService(Context.ACTIVITY_SERVICE);
+            ActivityManager.MemoryInfo memInfo = new ActivityManager.MemoryInfo();
+            activityManager.getMemoryInfo(memInfo);
+            memory = 1 + memInfo.totalMem / (1024 * 1024 * 1024);
+        } catch(Exception e){
+            System.out.println("In getDeviceRAM error: ");
+            e.printStackTrace();
+        }
+
+        if(memory == 0)
+            deviceRAM = "null";
+        else deviceRAM = memory+" GB";
+
+        sharedPref.edit().putString("DEVICE_RAM", deviceRAM).apply();
+
+        return deviceRAM;
+    }
+    public String[] getScreenDimensions()
+    {
+
+        String[] res= {sharedPref.getString("DEVICE_RESOLUTION", "__failed"),sharedPref.getString("DEVICE_SIZE", "__failed")};
+        
+        if(res[0] != "__failed" && res[1] != "__failed")
+            return res;
+
+        int height = 0;
+        int width  = 0;
+        float size = 0;
+
+        try {
+            DisplayMetrics displayMetrics = new DisplayMetrics();
+            getWindowManager().getDefaultDisplay().getRealMetrics(displayMetrics);
+            height = displayMetrics.heightPixels;
+            width = displayMetrics.widthPixels;
+            float x = height / displayMetrics.ydpi;
+            float y = width / displayMetrics.xdpi;
+            size = (float) Math.sqrt(x * x + y * y);
+            size = Math.round(size * 100) / 100;
+            
+        }catch(Exception e){
+            System.out.println("In getScreenDimensions error: ");
+            e.printStackTrace();
+        }
+
+        if(height != 0 && width != 0)
+        {
+            res[0]= height+"x"+width+"px";
+        }
+        else res[0] = "null";
+
+        if( size != 0 )
+            res[1] = size + " Inches";
+        else res[1] = "null";
+
+        sharedPref.edit().putString("DEVICE_RESOLUTION", res[0]).apply();
+        sharedPref.edit().putString("DEVICE_SIZE", res[1]).apply();
+
+        return res;
+    }
+
+    public String getDeviceDetails()
+    {
+        String deviceDetails = "";
+        try {
+            String bVersion = Build.VERSION.RELEASE;
+            String bModel = Build.MODEL;
+            String bBrand = Build.BRAND;
+            String[] dim = getScreenDimensions();
+            String deviceRAM = getDeviceRAM();
+
+            if(bModel == null || bModel == "")
+                bModel="null";
+            if(bBrand == null || bBrand == "")
+                bBrand="null";
+            if(bVersion == null || bVersion == "")
+                bVersion = "null";
+            else
+                bVersion = "Android v"+bVersion;
```

Comment:
use ternary operator here

Reply:
other places as well where you can optimize

Reply:
done